seqr/views/utils/anvil_metadata_utils.py

```diff
@@ -543,10 +544,52 @@ def _get_variant_inheritance(individual, genotypes):
 LIST_SAMPLE_FIELDS = ['SequencingProduct', 'dbgap_submission']
 
 
-def _get_sample_airtable_metadata(sample_ids, user, fields):
-    sample_records, _ = get_airtable_samples(
-        sample_ids, user, fields=fields or SINGLE_SAMPLE_FIELDS, list_fields=None if fields else LIST_SAMPLE_FIELDS,
+def _get_airtable_samples_for_id_field(sample_ids, id_field, fields, session):
+    raw_records = session.fetch_records(
+        'Samples', fields=[id_field] + fields,
+        or_filters={f'{{{id_field}}}': sample_ids},
     )
+
+    records_by_id = defaultdict(list)
+    for airtable_id, record in raw_records.items():
+        records_by_id[record[id_field]].append({**record, 'airtable_id': airtable_id})
+    return records_by_id
+
+
+def _get_sample_airtable_metadata(sample_ids, user, airtable_fields):
+    fields, list_fields = airtable_fields or [SINGLE_SAMPLE_FIELDS, LIST_SAMPLE_FIELDS]
+    all_fields = fields + list_fields
+
+    session = AirtableSession(user)
+    records_by_id = _get_airtable_samples_for_id_field(sample_ids, 'CollaboratorSampleID', all_fields, session)
+    missing = set(sample_ids) - set(records_by_id.keys())
+    if missing:
+        records_by_id.update(_get_airtable_samples_for_id_field(missing, 'SeqrCollaboratorSampleID', all_fields, session))
+
+    sample_records = {}
+    for record_id, records in records_by_id.items():
+        parsed_record = {}
+        for field in fields:
+            record_field = {
+                record[field][0] if field == 'Collaborator' else record[field] for record in records if field in record
+            }
+            if len(record_field) > 1:
+                error = 'Found multiple airtable records for sample {} with mismatched values in field {}'.format(
+                    record_id, field)
+                raise ErrorsWarningsException([error])
+            if record_field:
+                parsed_record[field] = record_field.pop()
+        for field in list_fields:
+            parsed_record[field] = {} if airtable_fields else set()
+            for record in records:
+                if field in record:
+                    if airtable_fields:
+                        parsed_record[field][record['airtable_id']] = record[field]
```

Comment:
this line is new, and aggregates list fields into a dictionary keyed by the original record id (so here SMID would be a dictionary of sample record IDs to the actual SMIDs. The default behavior for list fields is the below line, which remains unchanged